alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -494,6 +507,30 @@ class TransportIndexMonitorAction @Inject constructor(
             if (request.monitor.enabled && currentMonitor.enabled)
                 request.monitor = request.monitor.copy(enabledTime = currentMonitor.enabledTime)
 
+            // On update monitor check which backend roles to associate to the monitor
+            if (user != null) {
+                if (!request.rbacRoles.isNullOrEmpty()) {
```

Comment:
Lets treat null and empty cases separately - 
Empty case is an intentional behavior and should be treated as if they want to remove all roles from a monitor which they can assume. 
Lets also add unit test for it.

Reply:
separated the cases